packages/core/src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -779,32 +787,107 @@ export class AbstractQueryGeneratorTypeScript {
   }
 
   bulkDeleteQuery(tableName: TableNameOrModel, options: BulkDeleteQueryOptions): string {
-    const table = this.quoteTable(tableName);
-    const whereOptions = isModelStatic(tableName) ? { ...options, model: tableName } : options;
+    return joinSQLFragments([
+      `DELETE FROM ${this.quoteTable(tableName)}`,
+      this.#internals.addWhereSelectLimitOffset(tableName, options),
+    ]);
+  }
 
-    if (options.limit && this.dialect.supports.delete.modelWithLimit) {
-      if (!isModelStatic(tableName)) {
-        throw new Error('Cannot use LIMIT with bulkDeleteQuery without a model.');
+  /**
+   * Returns an update query
+   *
+   * @param tableName
+   * @param attrValueHash
+   * @param options
+   * @param columnDefinitions
+   */
+  updateQuery(
+    tableName: TableNameOrModel,
+    attrValueHash: Record<string, unknown>,
+    options?: UpdateQueryOptions,
+    columnDefinitions?: Record<string, AttributeOptions>,
+  ): QueryWithBindParams {
+    if (options) {
+      const UPDATE_QUERY_SUPPORTED_OPTIONS = new Set<keyof UpdateQueryOptions>();
+
+      if (this.dialect.supports.update.ignore) {
+        UPDATE_QUERY_SUPPORTED_OPTIONS.add('ignoreDuplicates');
       }
 
-      const pks = Object.values(tableName.primaryKeys).map(key => this.quoteIdentifier(key.columnName)).join(', ');
-      const primaryKeys = Object.values(tableName.primaryKeys).length > 1 ? `(${pks})` : pks;
+      if (this.dialect.supports.update.returning) {
+        UPDATE_QUERY_SUPPORTED_OPTIONS.add('returning');
+      }
 
-      return joinSQLFragments([
-        `DELETE FROM ${table} WHERE ${primaryKeys} IN (`,
-        `SELECT ${pks} FROM ${table}`,
-        options.where ? this.whereQuery(options.where, whereOptions) : '',
-        `ORDER BY ${pks}`,
-        this.#internals.addLimitAndOffset(options),
-        ')',
-      ]);
+      rejectInvalidOptions(
+        'updateQuery',
+        this.dialect.name,
+        UPDATE_QUERY_SUPPORTABLE_OPTIONS,
+        UPDATE_QUERY_SUPPORTED_OPTIONS,
```

Comment:
You can now pass a map of option name to a boolean directly: https://github.com/sequelize/sequelize/pull/17073